static/src/javascripts/projects/commercial/modules/article-body-adverts-wide.js

```diff
@@ -0,0 +1,241 @@
+    'common/modules/commercial/dfp/create-slot',
+    'common/modules/commercial/commercial-features',
+    'common/modules/onward/geo-most-popular',
+    'lodash/functions/memoize'
+], function (
+    Promise,
+    qwery,
+    stickyMpu,
+    config,
+    detect,
+    fastdom,
+    spaceFiller,
+    adSizes,
+    addSlot,
+    trackAdRender,
+    createSlot,
+    commercialFeatures,
+    mostPopular,
+    memoize
+) {
+    /* We keep a handle on the main column to compute offsets afterwards */
+    var mainColumn = qwery('.js-content-main-column')[0];
+
+    /* We keep track of inline MPUs so that we can offset them to the right later */
+    var inlineSlots = [];
+
+    /* The promise resolves either when an inline merch slot has been added and
+       a DFP call has returned, or directly if no inline slot has been added */
+    var waitForMerch = memoize(function (imSlot) {
+        return imSlot ? trackAdRender('dfp-ad--im') : Promise.resolve(true);
+    });
+
+    /* bodyAds is a counter that keeps track of the number of inline MPUs
+     * inserted dynamically. */
+    var bodyAds;
+
+    var isWide;
+    var isMobile;
+    var replaceTopSlot;
+    var getSlotName;
+
+    return {
+        init: init,
+
+        '@@tests': {
+            waitForMerch: waitForMerch
+        }
+    };
+
+    function init() {
+        if (!commercialFeatures.articleBodyAdverts) {
+            return Promise.resolve(false);
+        }
+
+        boot();
+
+        if (config.page.hasInlineMerchandise) {
+            var im = addInlineMerchAd();
+            im.then(waitForMerch).then(addInlineAds);
+            return im;
+        }
+
+        addInlineAds();
+
+        return Promise.resolve(true);
+    }
+
+    function boot() {
+        bodyAds = 0;
+        isWide = detect.getBreakpoint() === 'wide';
+        replaceTopSlot = isMobile = !isWide && detect.isBreakpoint({ max: 'tablet' });
+        getSlotName = replaceTopSlot ? getSlotNameForMobile : getSlotNameForDesktop;
+    }
+
+    function getSlotNameForMobile() {
+        bodyAds += 1;
+        return bodyAds === 1 ? 'top-above-nav' : 'inline' + (bodyAds - 1);
+    }
+
+    function getSlotNameForDesktop() {
+        bodyAds += 1;
+        return 'inline' + bodyAds;
+    }
+
+    function getRules(isMerch) {
+        var prevSlot;
+        var rules = {
+            bodySelector: '.js-article__body',
+            slotSelector: ' > p',
+            minAbove: isMobile ? 300 : config.page.hasShowcaseMainElement ? 900 : 700,
+            minBelow: adSizes.mpu.height,
+            selectors: {
+                ' > h2': {minAbove: isMobile ? 100 : 0, minBelow: 250},
+                ' .ad-slot': {minAbove: 500, minBelow: 500},
+                ' > :not(p):not(h2):not(.ad-slot)': {minAbove: 35, minBelow: 400}
+            },
+            filter: function(slot) {
+                if (!prevSlot || Math.abs(slot.top - prevSlot.top) - adSizes.mpu.height >= rules.selectors[' .ad-slot'].minBelow) {
```

Comment:
Yep, that is the minimum space between two ad slots. This is only necessary when (1) there is an inline merch slot, or (2) when we insert MPUs during the second call ("long article ads")